### PR TITLE
allow empty max ping filter

### PR DESCRIPTION
**Pull request by BannZay**
The inconvenieces I discovered so far:
1. You got to write 999 to "disable" this filter
2. If you try to clear this field and then type 75 then the field will contain 750 because the last zero is bugged at this point
Those are fixed

What I would like to be fixed as well (not sure if I gona do this soon):
1. Update browser servers live without refreshing. You press refresh - you receives all the servers and then you should be able (but currently not) to filter them out without pressing on "Refresh" button again but just by changing the max ping value

### CODE REVIEW

File: src/game/client/components/menus_browser.cpp [hunk 1]
```diff
@@ -614,8 +614,9 @@ void CMenus::RenderServerbrowserFilters(CUIRect View)
 
 		UI()->DoLabelScaled(&Button, Localize("Maximum ping:"), FontSize, -1);
 
-		char aBuf[5];
-		str_format(aBuf, sizeof(aBuf), "%d", g_Config.m_BrFilterPing);
+		char aBuf[5] = "";
+		if (g_Config.m_BrFilterPing != 0)
```
Comment (heinrich5991) [inline src/game/client/components/menus_browser.cpp:618]:
```suggestion
		if(g_Config.m_BrFilterPing != 0)
```

**Comment by heinrich5991** [comment]
>     1. Update browser servers live without refreshing. You press refresh - you receives all the servers and then you should be able (but currently not) to filter them out without pressing on "Refresh" button again but just by changing the max ping value

That doesn't work? Huh. It seems to work when decreasing the ping, at least.

**Comment by heinrich5991** [comment]
bors r+

Thanks.

**Comment by BannZay** [comment]
> > 1. Update browser servers live without refreshing. You press refresh - you receives all the servers and then you should be able (but currently not) to filter them out without pressing on "Refresh" button again but just by changing the max ping value
> 
> That doesn't work? Huh. It seems to work when decreasing the ping, at least.

If you just change the numbers in the field it does not (tested on latest released, master)
I guess you tested it wrong. Probably you changed the number while your server list being updated. If so - it does updates everything every time new server info received so yes in this case it works

**Comment by heinrich5991** [comment]
bors retry

> I guess you tested it wrong. Probably you changed the number while your server list being updated (tab changed or button refresh recently pressed). If so - it does updates everything every time new server info received so yes in this case it works

Ah yea, I see.

**Pull request merged**